mobile/android/android-emulator-services.ts

```diff
@@ -120,15 +152,124 @@ class AndroidEmulatorServices implements Mobile.IEmulatorPlatformServices {
 		return timeout * 1000;
 	}
 
+	private listVirtualMachines(parameter: string): IFuture<VirtualMachine[]> {
+		var future = new Future<VirtualMachine[]>();
+		var allData: string = "";
+		var virtualMachines: VirtualMachine[] = [];
+
+		var vboxPath = path.join(this.getPathToVBox().wait(), "vboxmanage");
+		var vboxManage = this.$childProcess.spawn(vboxPath, ["list", parameter]);
+
+		vboxManage.stdout.on("data", (chunk: string) => {
+			if(chunk) {
+				allData += chunk;
+			}
+		});
+		vboxManage.stderr.on("data", (error: Error) => {
+			// maybe use log trace level to print the error
+			this.$errors.fail("Unable to get currently running virtual machines. Check if VirtualBox is installed correctly.");
+		});
+
+		vboxManage.on("exit", () => {
```

Comment:
I suggest you to use`var childProcess = this.$childProcess.spawnFromEvent(vboxPath, ["list", parameter], "exit", {throwError: false}).wait(); 
if(childProcess. stderr) {
  // maybe use log trace level to print the error
  this.$errors.fail("Unable to get currently running virtual machines. Check if VirtualBox is installed correctly.");
}`